R/PredictionSurv.R

```diff
@@ -148,9 +158,16 @@ PredictionSurv = R6Class("PredictionSurv",
         surv
       }
     },
-    .distrify_survmatrix = function(x) {
-      distr6::as.Distribution(1 - x, fun = "cdf",
-        decorators = c("CoreStatistics", "ExoticStatistics"))
+    .distrify_survarray = function(x) {
+      if (inherits(x, "matrix")) {
+        # create Matdist
+        distr6::as.Distribution(1 - x, fun = "cdf",
+          decorators = c("CoreStatistics", "ExoticStatistics"))
+      } else {
+        # create Arrdist
+        distr6::Arrdist$new(cdf = 1 - x, which.curve = "mean",
+          decorators = c("CoreStatistics", "ExoticStatistics"))
+      }
```

Comment:
Not sure why this was needed, should've worked as it was

Reply:
I wrote it in the chat => it is because with `x` a survival array the following doesn't work:
```
distr6::as.Distribution(1 - x, fun = "cdf", which.curve = "mean",
          decorators = c("CoreStatistics", "ExoticStatistics"))
```

Reply:
Right because it's median by default, but median is sometimes better established than mean for Bayesian methods no?